trident/absorption_spectrum/absorption_spectrum.py

```diff
@@ -740,17 +749,30 @@ def _add_lines_to_spectrum(self, field_data, use_peculiar_velocity,
             n_absorbers = len(lambda_obs)
 
             # thermal broadening b parameter
-            thermal_b =  np.sqrt((2 * boltzmann_constant_cgs *
-                                  field_data['temperature']) /
-                                  line['atomic_mass'])
+            inv_temp = 1. / field_data['temperature'].d
+
+            if line['wavelength'] == 2.1e9:
+                thermal_b = np.ones(len(redshift)) * YTQuantity(self.bin_width.d,'km/s')
```

Comment:
Setting the thermal_b to bin_width. Makes sure that absorbers inside a given bin stay inside a bin, atleast in velocity space.